From f13deee1a171447c4747d2c6d21bc4607769879e Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Sun, 18 Apr 2021 15:57:15 +0200
Subject: [PATCH 47/50] Fix crash on start, systemReady might be called too
 late to get mLights

Regression caused by I6094c446e0b8c23f57d30652a3cbd35dee5e821a
Fix crash on start, systemReady might be called too late to get mLights

Regression caused by I6094c446e0b8c23f57d30652a3cbd35dee5e821a
---
 .../java/com/android/server/power/PowerManagerService.java  | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/services/core/java/com/android/server/power/PowerManagerService.java b/services/core/java/com/android/server/power/PowerManagerService.java
index f7a7d1829dc3..b5117834f640 100644
--- a/services/core/java/com/android/server/power/PowerManagerService.java
+++ b/services/core/java/com/android/server/power/PowerManagerService.java
@@ -2298,9 +2298,9 @@ public final class PowerManagerService extends SystemService
                     if (now < nextTimeout) {
                         if (getWakefulnessLocked() == WAKEFULNESS_AWAKE) {
                             if (now > mLastUserActivityTime + BUTTON_ON_DURATION) {
-                                mButtonsLight.setBrightness(PowerManager.BRIGHTNESS_OFF_FLOAT);
+                                if(mButtonsLight != null) mButtonsLight.setBrightness(PowerManager.BRIGHTNESS_OFF_FLOAT);
                             } else {
-                                mButtonsLight.setBrightness(mScreenBrightnessSettingDefault);
+                                if(mButtonsLight != null) mButtonsLight.setBrightness(mScreenBrightnessSettingDefault);
                                 nextTimeout = now + BUTTON_ON_DURATION;
                             }
                         }
@@ -2310,7 +2310,7 @@ public final class PowerManagerService extends SystemService
                         if (now < nextTimeout) {
                             mUserActivitySummary = USER_ACTIVITY_SCREEN_DIM;
                             if (getWakefulnessLocked() == WAKEFULNESS_AWAKE) {
-                                mButtonsLight.setBrightness(PowerManager.BRIGHTNESS_OFF_FLOAT);
+                                if(mButtonsLight != null) mButtonsLight.setBrightness(PowerManager.BRIGHTNESS_OFF_FLOAT);
                             }
                         }
                     }
-- 
2.25.1

